### PR TITLE
Added 3d Jupyter Notebook Viewer

### PR DESCRIPTION
**Pull request by deeptavker**
The 3D viewer currently supports:

1) ipyvolume scatter (any output property) and quiver plots (for <u,v,w> )
2) Changeable colormaps for scatter plots and corresponding legends
3) Changeable glyph sizes for both scatter and vector plots 
----------

ui3d620
uixyz
ui3d3dui

--------------

### CODE REVIEW

File: pysph/tools/ipy_viewer.py [hunk 1]
```diff
@@ -540,3 +541,294 @@ def _save_figure_handler(self, change):
...
+
+
+class ParticleArrayWidgets3D(object):
+
+    def __init__(self, particlearray, particles_type):
+        self.particles_type = particles_type
+        self.scalar = widgets.Dropdown(
+            options=[
+                'None'
+            ]+particlearray.output_property_arrays,
```
Comment (prabhuramachandran) [inline pysph/tools/ipy_viewer.py]: PEP8, space around +

File: pysph/tools/ipy_viewer.py [hunk 2]
```diff
@@ -540,3 +541,294 @@ def _save_figure_handler(self, change):
...
+            pa_widgets.scalar_cmap.observe(self._scalar_cmap_handler, 'value')
+
+    def interactive_plot(self):
+        self._create_widgets()
+        self.scatters = {}
+        display(self._widgets._create_vbox())
+        self.vectors = {}
+        self.legend = widgets.Output()
+
+        import ipyvolume.pylab as p3
```
Comment (prabhuramachandran) [inline pysph/tools/ipy_viewer.py]: Why the name `p3`?

File: pysph/tools/ipy_viewer.py [hunk 3]
```diff
@@ -540,3 +541,294 @@ def _save_figure_handler(self, change):
                 )
                 break
         self._widgets.save_figure.value = ""
+
+
+class ParticleArrayWidgets3D(object):
+
+    def __init__(self, particlearray, particles_type):
```
Comment (prabhuramachandran) [inline pysph/tools/ipy_viewer.py]: what is `particles_type`?  If you mean or the `name` of the array?

Comment (deeptavker) [reply pysph/tools/ipy_viewer.py]: It's the same as we were using for 2d. It takes the values - _fluid_, _boundary_, _inlet_, etc.

File: pysph/tools/ipy_viewer.py (hunk 2 above)
Comment (deeptavker) [reply pysph/tools/ipy_viewer.py]: I saw this in a couple of examples.

File: pysph/tools/ipy_viewer.py (hunk 1 above)
Comment (deeptavker) [reply pysph/tools/ipy_viewer.py]: Yep, I wonder why pep8 did not catch that.

File: pysph/tools/ipy_viewer.py (hunk 3 above)
Comment (prabhuramachandran) [reply pysph/tools/ipy_viewer.py]: Then I would prefer that you change both the variable names to something more meaningful like `array_name` or even just `name`.  That said, you don't need to pass the attribute at all as the name is always available from the particle array itself.  `particlearray.name` will give you the same thing.  Can you please fix this?  Thanks.

**Comment by prabhuramachandran** [comment]
@deeptavker -- I hope you have tested these out once as we do not have any tests for this.  Actually, it may be nice to have some tests for this eventually, perhaps using mocks.  If you can confirm that these all work, I am happy to merge for now but do keep the testing in mind for later.

**Comment by deeptavker** [comment]
Tested, RTM. I'll have a look at the mock tests after Binder and documentation.

**Pull request merged**